Libraries/LibWeb/Bindings/MainThreadVM.cpp

```diff
@@ -471,62 +471,64 @@ ErrorOr<void> initialize_main_thread_vm(HTML::EventLoop::Type type)
         }
 
         // 7. If referrer is a Cyclic Module Record and moduleRequest is equal to the first element of referrer.[[RequestedModules]], then:
-        if (referrer.has<GC::Ref<JS::CyclicModule>>() && module_request == referrer.get<GC::Ref<JS::CyclicModule>>()->requested_modules().first()) {
-            // 1. For each ModuleRequest record requested of referrer.[[RequestedModules]]:
-            for (auto const& module_request : referrer.get<GC::Ref<JS::CyclicModule>>()->requested_modules()) {
-                // 1. If moduleRequest.[[Attributes]] contains a Record entry such that entry.[[Key]] is not "type", then:
-                for (auto const& attribute : module_request.attributes) {
-                    if (attribute.key == "type"sv)
-                        continue;
+        if (referrer.has<GC::Ref<JS::CyclicModule>>()) {
+            if (auto const& requested_modules = referrer.get<GC::Ref<JS::CyclicModule>>()->requested_modules(); !requested_modules.is_empty() && module_request == requested_modules.first()) {
```

Comment:
The spec doesn't seem to read to me that this could be empty 🤔  Maybe somewhere else is going wrong

Reply:
Yeah needs more debooging, but let's keep things working at least :)